File: 1496-path-crossing/1496-path-crossing.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        int cx = 0, cy = 0;
        
        set<pair<int, int>>seen;
        seen.insert({cx, cy});
        
        for(char dir : path){
            
            if(dir == 'N')  ++cy;
            
            else if(dir == 'S')     --cy;
            
            else if(dir == 'E')     ++cx;
            
            else    --cx;
            
            if(seen.find({cx, cy}) != seen.end())   return true;
            
            seen.insert({cx, cy});
            
        }
        
        return false;
        
    }
};

File: 1496-path-crossing/1496-path-crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
        int x = 0, y = 0;
        
        Set<String> seen = new HashSet<>();
        seen.add("0,0");
        
        for(char dir : path.toCharArray()){
            
            if(dir == 'N')  ++y;
            
            else if(dir == 'S')     --y;
            
            else if(dir == 'E')     ++x;
            
            else    --x;
            
            String res = String.valueOf(x) + "," + String.valueOf(y);
            
            if(seen.contains(res))  return true;
            
            seen.add(res);
        }
        
        return false;
    }
}